### PR TITLE
Fixed #28865 -- Updated .py from .pyc calculation.

### PR DESCRIPTION
**Pull request by ramiro**
Made code in the autoreloader be compatible with Python 3.
https://code.djangoproject.com/ticket/28865

### CODE REVIEW

**Comment by ramiro** [comment]
Ticket: https://code.djangoproject.com/ticket/28865

**Comment by timgraham** [comment]
This might be obsoleted (if it's not already, as you noted you couldn't figure out if this code ever executes) by https://github.com/django/django/pull/8819. Perhaps we're better off investing time there? \cc @orf.

**Comment by orf** [comment]
I think it's worth spending time in #8819, I'm going to spend some time today to clear up the PR (it's a mess right now). I also couldn't work out when this was executed, IMO unless we hear a convincing reason for it to be there lets remove it.

**Comment by orf** [comment]
I'm struggling to think of one anyway, if you're loading `pyc` files into sys.path it would mean the `.py` file is not there, in which case the `pyc` file wouldn't be in `__pycache__` anyway. But if that's the case then they wouldn't change, so wouldn't have to be autoreloaded. Unless you're doing something *very very* strange.

File: tests/utils_tests/test_autoreload.py [hunk 1]
```diff
@@ -200,7 +200,20 @@ class CleanFilesTests(SimpleTestCase):
...
             [None, 'file1.pyo', 'file2.pyc', 'myclass$py.class'],
             ['file1.py', 'file2.py', 'myclass.py'],
-        )
+        ),
+        'with_path': ([os.path.join('django', 'utils', 'file.pyc')], [os.path.join('django', 'utils', 'file.py')]),
+        'python3': (
+            [
+                os.path.join('django', 'utils', '__pycache__', 'file1.cpython-36.pyc'),
+                os.path.join(os.path.sep, '__pycache__', 'file2.cpython-36.pyc'),
+                os.path.join('__pycache__', 'file3.cpython-36.pyc')
```
Comment (mimi1vx) [inline tests/utils_tests/test_autoreload.py]: looks like python-3.6 only code:) it will probably fail on all other releases

**Pull request closed without merging**